### PR TITLE
build(docker): Introduce a docker and docker-compose setup

### PR DESCRIPTION
**Pull request by MichaelsJP**

### CODE REVIEW

**Comment by MichaelsJP** [comment]
Waiting for #284

**Comment by MichaelsJP** [comment]
closes  #291

File: README.md [hunk 1]
```diff
@@ -78,16 +67,76 @@ pnpm install
...
+docker build --tag ors-map-client:local .
+```
+
+2. Run a container:
+
+```shell
+docker run -d -it -v "$(pwd)"/nginx/logs:/var/log/nginx:rw -p 127.0.0.1:8080:80 --name ors-map-client ors-map-client:local
+```
+
+3. Go to http://127.0.0.1:8080 and enjoy!
```
Comment (TheGreatRefrigerator) [inline README.md:96]: Not sure about the `and enjoy!` 😅 part 
would put like:
`The app should be running at ...`

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
I would also split between like:
- feat: Add docker & docker compose
- docs: Add instructions for docker (compose) workflows
- feat: Add docker hub checks / deployment

**Comment by TheGreatRefrigerator** [comment]
Also still need to test locally

File: README.md [hunk 2]
```diff
@@ -78,16 +67,76 @@ pnpm install
...
+```shell
+docker-compose up -d
+```
+
+2. Go to http://127.0.0.1:8080 and enjoy!
+
+If you want to check the logs, run:
+
+```shell
+docker-compose logs -ft
```
Comment (TheGreatRefrigerator) [inline README.md:119]: should we use the `docker compose` instead or is `docker-compose` (standalone thing) still the default?
I think compose comes with all Docker Desktop versions available as `docker compose`

File: README.md (hunk 1 above)
Comment (TheGreatRefrigerator) [reply README.md:96]: Ah and do you think we should add commands to stop/remove the container again for `docker` and `docker compose`?
And on the same thought... is there a benefit of using the native docker command?
Or should we just remove that part and only show the docker compose, which is the most simple deployment?

**Comment by TheGreatRefrigerator** [comment]
Both deployments for `docker` and `docker compose` worked flawlessly in a local test 👍🏻

File: README.md (hunk 1 above)
Comment (MichaelsJP) [reply README.md:96]: > Ah and do you think we should add commands to stop/remove the container again for `docker` and `docker compose`? And on the same thought... is there a benefit of using the native docker command? Or should we just remove that part and only show the docker compose, which is the most simple deployment?

I'd say both. Not everyone uses compose. Moreover, sometimes for a quick and dirty run, it's worth having a docker one-liner.

File: README.md [hunk 3]
```diff
@@ -522,6 +522,4 @@ This approach is intended to create cohesion and keep the project sustainable.
 There are additional documents that are part of the software documentation. they are in the folder `/docs` and are listed below:
 
 - [docs/dynamic-inputs.md](docs/dynamic-inputs.md) - describe how the inputs are rendered using a custom engine and not hard-coded
-- [docs/learned-lessons.md](docs/learned-lessons.md) - lessons learned and that might help in the future
-- [docs/learned-lessons.md](docs/learned-lessons.md) - explains what are the criteria for the search results
```
Comment (TheGreatRefrigerator) [inline README.md:488]: There is the file `search-results-criteria.md` which the second link should go to.
Removal of the first one is fine (although not really part of the docker change :P but we can make an exception)

File: README.md [hunk 4]
```diff
@@ -78,16 +67,65 @@ pnpm install
...
 - `bitlyLogin` - the Bitly login (used to shorten the share URL)
 
-5. The ORS menu is loaded/used by default. If you want to use a custom menu, have a look in the `hooks-example.js`.
+3. The ORS menu is loaded/used by default. If you want to use a custom menu, have a look in the `hooks-example.js`.
 
 The filters, theme and hooks of the map client can be customized if needed.
 
+##### Run with `Docker`
+
+Run the app in `docker`:
```
Comment (TheGreatRefrigerator) [inline README.md:82]: check [markdown rendering on this branch](https://github.com/GIScience/ors-map-client/tree/291-dockerize-the-application#run-with-docker)
would remove - no additional value

File: README.md [hunk 5]
```diff
@@ -78,16 +67,65 @@ pnpm install
...
+
+# Check the logs with
+docker logs --follow ors-map-client
+```
+
+The app should now be running at http://127.0.0.1:8080.
+
+##### Run with `docker compose`
+
+Run the app in `docker compose`:
```
Comment (TheGreatRefrigerator) [inline README.md:99]: see reason above - would remove

File: README.md [hunk 6]
```diff
@@ -78,16 +67,65 @@ pnpm install
...
+
+```shell
+# Build and run the image
+docker-compose up -d
+
+# Check the logs with
+docker-compose logs -ft
```
Comment (TheGreatRefrigerator) [inline README.md:106]: just for consistency with Header, either use `docker-compose` only or `docker compose` only.

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
few small changes that can be added to the related commits (create new with changes and squash to related, or via interactive rebase)

**Review by TheGreatRefrigerator — APPROVED** [review]
Looks good

**Pull request merged**